CMakeLists.txt

```diff
@@ -137,6 +139,12 @@ endif ()
 target_compile_definitions (OpenCL
   PRIVATE
     CL_TARGET_OPENCL_VERSION=300
+    CL_ICDL_OCL_VERSION_STRING="OpenCL 3.0"
```

Comment:
Does it make sense to build this string similar to the OPENCL_ICD_LOADER_VERSION_STRING as "OpenCL " + major version + "." minor version vs. passing it explicitly?  This could be important if we drive more versioning from the CMake project version (#182).

Reply:
Yes, I see your point. I'll look into it.

Reply:
I pushed https://github.com/KhronosGroup/OpenCL-ICD-Loader/pull/188/commits/01c9c3136ec26a10630ecca4da2ffadde1a42ac4 that partially address this comment. `OPENCL_ICD_LOADER_OCL_VERSION_STRING` is really a reflect of the `CL_TARGET_OPENCL_VERSION` used to compile the loader and the API exported. The issue is OpenCL does not define MINOR/MAJOR macros that we could use to get the string from `CL_TARGET_OPENCL_VERSION`. We could experiment in the `loader/icd_version.h` file and maybe provide those in the official headers once we are satisfied.

Reply:
Yes you could build a recent loader ( layers, trace, etc) but target OpenCL 2.0. It's version would still be 3.0.4, but the OCL_VERSION would be 2.0

Reply:
To expand a bit this is the reason for the different VERSION constants:
```C
#define CL_ICDL_OCL_VERSION 1
#define CL_ICDL_VERSION     2
```
The first is the OpenCL version supported, the second the version of the loader. For ocl-icd, the first returns "OpenCL 3.0" while the second returns "2.3.1"

Reply:
Makes sense, thanks for the explanation!